core/src/test/java/org/apache/accumulo/core/util/compaction/CompactionServicesConfigTest.java

```diff
@@ -0,0 +1,109 @@
+  private final Property newPrefix = Property.COMPACTION_SERVICE_PREFIX;
+
+  @Test
+  public void testCompactionProps() {
+    ConfigurationCopy conf = new ConfigurationCopy();
+
+    conf.set("compaction.major.service.default.planner", DefaultCompactionPlanner.class.getName());
+    conf.set("compaction.major.service.default.planner.opts.maxOpen", "10");
+    conf.set("compaction.major.service.default.planner.opts.executors",
+        "[{'name':'small','type':'internal','maxSize':'32M','numThreads':2},{'name':'medium','type':'internal','maxSize':'128M','numThreads':2},{'name':'large','type':'internal','numThreads':2}]");
+
+    conf.set(oldPrefix.getKey() + "default.planner.opts.ignoredProp", "1");
+    conf.set(newPrefix.getKey() + "default.planner.opts.validProp", "1");
+
+    var compactionConfig = new CompactionServicesConfig(conf);
+    assertTrue(compactionConfig.getOptions().get("default").containsKey("validProp"));
+    assertEquals("1", compactionConfig.getOptions().get("default").get("validProp"));
+    assertNull(compactionConfig.getOptions().get("default").get("ignoredProp"));
+  }
+
+  @Test
+  public void testDuplicateCompactionPlannerDefs() {
+    ConfigurationCopy conf = new ConfigurationCopy();
+
+    String planner = DefaultCompactionPlanner.class.getName();
+    String oldPlanner = "OldPlanner";
+
+    conf.set(newPrefix.getKey() + "default.planner", planner);
+    conf.set(oldPrefix.getKey() + "default.planner", oldPlanner);
+
+    conf.set(oldPrefix.getKey() + "old.planner", oldPlanner);
+
+    var compactionConfig = new CompactionServicesConfig(conf);
+
+    assertTrue(compactionConfig.getPlanners().containsKey("default"));
+    assertEquals(planner, compactionConfig.getPlanners().get("default"));
+
+    assertTrue(compactionConfig.getPlanners().containsKey("old"));
+    assertEquals(oldPlanner, compactionConfig.getPlanners().get("old"));
+  }
+
+  @Test
+  public void testCompactionPlannerOldDef() {
+    ConfigurationCopy conf = new ConfigurationCopy();
+
+    conf.set(oldPrefix.getKey() + "cs1.planner", DefaultCompactionPlanner.class.getName());
+    conf.set(oldPrefix.getKey() + "cs1.planner.opts.maxOpen", "10");
+    conf.set(oldPrefix.getKey() + "cs1.planner.opts.executors",
+        "[{'name':'small','type':'internal','maxSize':'32M','numThreads':2},{'name':'medium','type':'internal','maxSize':'128M','numThreads':2},{'name':'large','type':'internal','numThreads':2}]");
+    conf.set(oldPrefix.getKey() + "cs1.planner.opts.foo", "1");
+    conf.set(newPrefix.getKey() + "cs1.planner.opts.bar", "2");
+
+    var compactionConfig = new CompactionServicesConfig(conf);
+    assertTrue(compactionConfig.getOptions().get("cs1").containsKey("foo"));
+    assertEquals("1", compactionConfig.getOptions().get("cs1").get("foo"));
+
+    assertTrue(compactionConfig.getOptions().get("cs1").containsKey("bar"));
+    assertEquals("2", compactionConfig.getOptions().get("cs1").get("bar"));
```

Comment:
I was thinking service config would only be pulled from a single prefix, not a mix of both.  Thinking in this case that since `newPrefix+"cs1"` exists that everything under `oldPrefix+"cs1"` should be ignored.  Does not have to be done this way, its just what I thought would be less confusing because if a user only looks at config under one prefix then they will know its everything.  When mixed if the config is sorted, then all the config for a service would not sort together (like in the shell).